### PR TITLE
new api endpoint for map

### PR DESCRIPTION
**Pull request by ntrncic**
fixes unicef/etools-issues#150

### CODE REVIEW

File: EquiTrack/partners/views/interventions_v2.py [hunk 1]
```diff
@@ -341,3 +342,33 @@ def delete(self, request, *args, **kwargs):
...
+class InterventionListMapView(ListCreateAPIView):
+    """
+    Create new Interventions.
+    Returns a list of Interventions.
+    """
+    serializer_class = InterventionListMapSerializer
+    permission_classes = (IsAdminUser,)
+
+    def get_queryset(self):
+        q = Intervention.objects.all()
```
Comment (robertavram) [inline EquiTrack/partners/views/interventions_v2.py]: prefetch related to the needed level...

File: EquiTrack/partners/views/interventions_v2.py [hunk 2]
```diff
@@ -341,3 +342,33 @@ def delete(self, request, *args, **kwargs):
...
+            queries = []
+            if "country_programme" in query_params.keys():
+                queries.append(Q(agreement__country_programme=query_params.get("country_programme")))
+            if "sector" in query_params.keys():
+                queries.append(Q(sector_locations__sector__id__in=[query_params.get("sector")]))
+            if "status" in query_params.keys():
+                queries.append(Q(status__in=[query_params.get("status")]))
+            if "partner" in query_params.keys():
+                queries.append(Q(agreement__partner__in=[query_params.get("partner")]))
+            if "donor" in query_params.keys():
```
Comment (robertavram) [inline EquiTrack/partners/views/interventions_v2.py]: not sure what this is.. if donor.. grant__gte=query_params.get("start").. can you comment in what this is doing?

**Comment by robertavram** [comment]
sorry just saw this was WIP

File: EquiTrack/partners/models.py [hunk 3]
```diff
@@ -1777,13 +1777,19 @@ def __unicode__(self):
...
+
+
 class InterventionSectorLocationLink(models.Model):
     intervention = models.ForeignKey(Intervention, related_name='sector_locations')
     sector = models.ForeignKey(Sector, related_name='intervention_locations')
     locations = models.ManyToManyField(Location, related_name='intervention_sector_locations', blank=True)
 
     tracker = FieldTracker()
-
+    objects = models.Manager()
```
Comment (robertavram) [inline EquiTrack/partners/models.py]: did you mean to say InterventionSectorLocationLinkManager ?

File: EquiTrack/partners/views/interventions_v2.py (hunk 1 above)
Comment (robertavram) [inline EquiTrack/partners/views/interventions_v2.py]: the prefetch related is probably better suited here than the manager

File: EquiTrack/locations/serializers.py [hunk 4]
```diff
@@ -32,6 +32,7 @@ class Meta:
 class LocationSerializer(serializers.ModelSerializer):
 
     id = serializers.CharField(read_only=True)
+    gateway = serializers.CharField(source='gateway.name')
```
Comment (robertavram) [inline EquiTrack/locations/serializers.py]: we're not using gateway anymore... that;s from EquiTrack times

File: EquiTrack/partners/models.py [hunk 5]
```diff
@@ -1216,12 +1216,12 @@ class InterventionManager(models.Manager):
     def get_queryset(self):
         return super(InterventionManager, self).get_queryset().prefetch_related('result_links',
                                                                                 'sector_locations__sector',
+                                                                                'sector_locations__locations',
```
Comment (robertavram) [inline EquiTrack/partners/models.py]: locations should be loaded from the locations endpoint to the frontend.. locations endpoint has the locations cached.

File: EquiTrack/locations/serializers.py (hunk 4 above)
Comment (robertavram) [inline EquiTrack/locations/serializers.py]: we're not using Gateway anymore.. EquiTrack times

File: EquiTrack/partners/serializers/interventions_v2.py [hunk 6]
```diff
@@ -468,3 +468,27 @@ class Meta:
             'id', 'number', 'partner_name', 'status', 'title', 'start', 'end',
             'government_intervention', 'planned_budget'
         )
+
+
+class InterventionLocationSectorMapNestedSerializer(serializers.ModelSerializer):
+    locations = LocationLightSerializer(many=True)
```
Comment (robertavram) [inline EquiTrack/partners/serializers/interventions_v2.py]: no need to serialize locations... they are in the frontend.. ids is enough

File: EquiTrack/partners/views/interventions_v2.py [hunk 7]
```diff
@@ -39,12 +39,14 @@
...
     InterventionSectorLocationCUSerializer,
-    InterventionResultCUSerializer
+    InterventionResultCUSerializer,
+    InterventionListMapSerializer,
 )
 from partners.exports_v2 import InterventionCvsRenderer
 from partners.filters import PartnerScopeFilter
 from partners.validation.interventions import InterventionValid
 from partners.permissions import PartneshipManagerRepPermission
+from locations.models import Location
```
Comment (robertavram) [inline EquiTrack/partners/views/interventions_v2.py]: why are we importing Location?

File: EquiTrack/locations/serializers.py (hunk 4 above)
Comment (ntrncic) [inline EquiTrack/locations/serializers.py]: Should I change the model then to `type` instead of `gateway`

Comment (ntrncic) [inline EquiTrack/locations/serializers.py]: changed to type on serialzier

File: EquiTrack/partners/serializers/interventions_v2.py (hunk 6 above)
Comment (ntrncic) [inline EquiTrack/partners/serializers/interventions_v2.py]: removed locations serializer just returning ids

File: EquiTrack/partners/views/interventions_v2.py (hunk 7 above)
Comment (ntrncic) [inline EquiTrack/partners/views/interventions_v2.py]: removed!

**Pull request merged**